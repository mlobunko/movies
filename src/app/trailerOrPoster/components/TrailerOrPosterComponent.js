import React from 'react';
import PropTypes from 'prop-types';

import {
  StyledPoster,
  StyledTrailer,
  StyledTrailerContainer,
  StyledPosterContainer,
  StyledTrailerContent,
  StyledPosterContent
} from './styles';

export const TrailerOrPosterComponent = ({ posterPath, trailerPath }) => (
  <React.Fragment>
    {trailerPath ? (
      <StyledTrailerContainer>
        <StyledTrailerContent>
          <StyledTrailer trailerPath={trailerPath} />
        </StyledTrailerContent>
      </StyledTrailerContainer>
    ) : posterPath ? (
      <StyledPosterContainer>
        <StyledPosterContent>
          <StyledPoster posterPath={posterPath} />
        </StyledPosterContent>
      </StyledPosterContainer>
    ) : null}
  </React.Fragment>
);

TrailerOrPosterComponent.displayName = 'TrailerOrPosterComponent';

export default TrailerOrPosterComponent;

TrailerOrPosterComponent.propTypes = {
  posterPath: PropTypes.string,
  trailerPath: PropTypes.string
};
