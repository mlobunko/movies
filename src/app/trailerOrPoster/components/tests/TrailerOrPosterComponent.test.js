import React from 'react';
import { shallow } from 'enzyme';

import { TrailerOrPosterComponent } from '../TrailerOrPosterComponent';

describe('render', () => {
  it('posterPath true', () => {
    const wrapper = shallow(<TrailerOrPosterComponent posterPath={'abc'} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('trailerPath true', () => {
    const wrapper = shallow(<TrailerOrPosterComponent trailerPath={'cba'} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('trailerPath false, posterPath false', () => {
    const wrapper = shallow(<TrailerOrPosterComponent />);
    expect(wrapper).toMatchSnapshot();
  });
});
