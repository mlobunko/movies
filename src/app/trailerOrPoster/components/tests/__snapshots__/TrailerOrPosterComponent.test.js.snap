// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render posterPath true 1`] = `
<Fragment>
  <StyledPosterContainer>
    <StyledPosterContent>
      <StyledPoster
        posterPath="abc"
      />
    </StyledPosterContent>
  </StyledPosterContainer>
</Fragment>
`;

exports[`render trailerPath false, posterPath false 1`] = `<Fragment />`;

exports[`render trailerPath true 1`] = `
<Fragment>
  <StyledTrailerContainer>
    <StyledTrailerContent>
      <StyledTrailer
        trailerPath="cba"
      />
    </StyledTrailerContent>
  </StyledTrailerContainer>
</Fragment>
`;
